src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -1236,23 +1552,12 @@ def _getRoleInstanceProfileNames(self, role_name: str) -> List[str]:
         Returns instance profile names.
         """
 
-        allProfiles = []
-
-        response = self.iam_client.list_instance_profiles_for_role(RoleName=role_name,
-                                                                   MaxItems=200)
-        while True:
-            # Process the current page
-            allProfiles += [item['InstanceProfileName'] for item in response.get('InstanceProfiles', [])]
-            if 'IsTruncated' in response and response['IsTruncated']:
-                # There are more pages. Get the next one, supplying the marker.
-                response = self.iam_client.list_instance_profiles_for_role(RoleName=role_name,
-                                                                           MaxItems=200,
-                                                                           Marker=response['Marker'])
-            else:
-                # No more pages
-                break
+        # Grab the connection we need to use for this operation.
+        iam = self.aws.client(self._region, 'iam')
 
-        return allProfiles
+        return [item['InstanceProfileName'] for item in self._pager(iam.list_instance_profiles_for_role,
```

Comment:
Same concern as above, though I guess we've never run into problems before, so maybe passing the whole list is always fine.

Reply:
There's probably exactly one item in this list all the time; we make the roles we use, and when the role is made it gets one instance profile added to it.

Reply:
You can specify your own `--awsEc2ProfileArn`, and maybe have it attached to a role that has a lot of instance profiles on it? But that seems like a pathological configuration we can support when a user actually has it.